### PR TITLE
Spaces image in delegate

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #10041

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14716/6/1

<details>


</details>

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-ocis</strong> https://drone.owncloud.com/owncloud/client/14716/7/1

<details>


</details>

File: src/libsync/graphapi/space.cpp [hunk 1]
```diff
@@ -44,7 +47,16 @@ OpenAPI::OAIDrive Space::drive() const
 void Space::setDrive(const OpenAPI::OAIDrive &drive)
 {
     _drive = drive;
-    Q_EMIT updated();
+    if (!imageUrl().isEmpty()) {
+        auto job = _spaceManager->account()->resourcesCache()->makeGetJob(imageUrl(), {}, this);
```
Comment (fmoc) [inline src/libsync/graphapi/space.cpp:51]: Still not a fan of this "lazy loading", but it'll do for now.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/libsync/graphapi/space.h [hunk 2]
```diff
@@ -48,9 +50,12 @@ namespace GraphApi {
...
         bool disabled() const;
 
-    Q_SIGNALS:
-        void updated();
+        /**
+         * The image url
+         */
+        QUrl imageUrl() const;
 
+        QIcon image() const;
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/libsync/graphapi/space.h:58]: Not a null icon but a fallback

**Review by fmoc — APPROVED** [review]

**Pull request merged**